gremlin-dotnet/src/Gremlin.Net/Driver/Connection.cs

```diff
@@ -194,6 +194,13 @@ private async Task SendMessagesFromQueueAsync()
                 }
             }
             Interlocked.CompareExchange(ref _writeInProgress, 0, 1);
+
+            // Since the loop ended and the write in progress was set to 0
+            // a new item could have been added, write queue can contain items at this time
+            if (!_writeQueue.IsEmpty && Interlocked.CompareExchange(ref _writeInProgress, 1, 0) == 1)
```

Comment:
Shouldn't this be `== 0` instead of `== 1`? We set `_writeInProgress` to `0` directly before and if another thread has already set this value to `1` in parallel then we don't need to call `SendMessagesFromQueueAsync` here as that thread is then already sending messages.

Reply:
Right, sorry, I just copy-pasted the suggested code.

Reply:
Fixed.